### PR TITLE
EventFinda API and some other bits

### PR DESCRIPTION
**Pull request by s3364859**
@6nop @s3364942 @s3494110 @s3482247 

In this PR:
 - EventFinda implementation
 - Moved a few shared types about
 - Added in some extra projects for parsing and such

events

### CODE REVIEW

File: CPT331.Data.Parsers/Parser.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿#region Using References
+
+using System;
+using System.Collections.Generic;
+
+#endregion
+
+namespace CPT331.Data.Parsers
+{
+	public class Parser
```
Comment (s3364859) [inline CPT331.Data.Parsers/Parser.cs]: Placeholder.

File: CPT331.Data.Parsers/SpreadsheetParser.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿#region Using References
+
+using System;
+using System.Collections.Generic;
+
+#endregion
+
+namespace CPT331.Data.Parsers
+{
+	public class SpreadsheetParser : Parser
```
Comment (s3364859) [inline CPT331.Data.Parsers/SpreadsheetParser.cs]: Placeholder.

File: CPT331.WebAPI.Parsers/EventFindaWebParser.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+using CPT331.Core.Extensions;
+using CPT331.Core.ObjectModel;
+
+#endregion
+
+namespace CPT331.WebAPI.Parsers
+{
+	public class EventFindaWebParser : WebParser
+	{
+		public EventFindaWebParser()
```
Comment (s3364859) [inline CPT331.WebAPI.Parsers/EventFindaWebParser.cs]: New type for consuming the EventFinda web API.

File: CPT331.WebAPI.Parsers/EventFindaWebParser.cs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+			List<EventInfo> events = new List<EventInfo>();
+
+			XmlNodeList xmlNodeList = xmlDocument.SelectNodes("/events/event");
+			xmlNodeList.OfType<XmlNode>().ToList().ForEach(m => events.Add(ToEvent(m)));
+
+			return events;
+		}
+
+		public List<EventInfo> GetEventsByLocation(string name, int postcode)
```
Comment (s3364859) [inline CPT331.WebAPI.Parsers/EventFindaWebParser.cs]: Placeholder.

**Comment by 6nop** [comment]
@s3364859 Carl
Presumably the Event classes are immutable because we are using a 3rd party API and cannot modify the data?

I'm guessing we'll implement getters for the crime data, so the data can be modified in the Admin MVC site?

**Pull request merged**